tripper/dataset/dataset.py

```diff
@@ -0,0 +1,942 @@
+"""Module for documenting datasets with Tripper.
+
+The dataset documentation follows the DCAT structure and is exposed as
+Python dicts with attribute access in this module.  This dict
+structure is used by the functions:
+  - `read_datadoc()`: Read documentation from YAML file and return it as dict.
+  - `save_dict()`: Save dict documentation to the triplestore.
+  - `load_dict()`: Load dict documentation from the triplestore.
+
+YAML documentation can also be stored directly to the triplestore with
+  - `save_datadoc()`: Save documentation from YAML file to the triplestore.
+
+For accessing and storing actual data, the following functions can be used:
+  - `load()`: Load documented dataset from its source.
+  - `save()`: Save documented dataset to a data resource.
+
+For searching the triplestore:
+  - `list_dataset_iris()`: Get IRIs of matching datasets.
+
+For interaction with OTEAPI:
+  - `get_partial_pipeline()`: Returns a OTELib partial pipeline.
+
+"""
+
+# pylint: disable=invalid-name,redefined-builtin,import-outside-toplevel
+import functools
+import io
+import json
+import re
+import secrets  # From Python 3.9 we could use random.randbytes(16).hex()
+import warnings
+from pathlib import Path
+from typing import TYPE_CHECKING
+from urllib.parse import urlparse
+
+import requests
+import yaml  # type: ignore
+
+from tripper import DCAT, OTEIO, RDF, Triplestore
+from tripper.utils import AttrDict, as_python
+
+if TYPE_CHECKING:  # pragma: no cover
+    from typing import Any, Iterable, List, Mapping, Optional, Sequence, Union
+
+    from tripper.utils import Triple
+
+# Cache decorator
+cache = (
+    functools.cache  # new in Python 3.9, smaller and faster than lru_cache()
+    if hasattr(functools, "cache")
+    else functools.lru_cache(maxsize=1)
+)
+
+
+# Pytest can't cope with this
+# EMMO = Namespace(
+#     iri="https://w3id.org/emmo#",
+#     label_annotations=True,
+#     check=True,
+# )
```

Comment:
What do we need this for? EMMO is defined as namespace here https://github.com/EMMC-ASBL/tripper/blob/1d38c2c8daafd9d4d63a099e9171cf49c9b2083f/tripper/namespace.py#L287

Reply:
The problem is that currently

```python
>>> from tripper import EMMO
>>> EMMO.Atom
'https://w3id.org/emmo#Atom'
```

which is not the correct IRI for the Atom class. It includes the numerical UUID. 

The above way to define the EMMO namespace produces the correct result, but doesn't work with pytest.